solutions/devsprint-1/DeliveryAppChallenge/Components/Cells/RestaurantCell.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  RestaurantCell.swift
+//  DeliveryAppChallenge
+//
+//  Created by Edson Neto on 04/11/21.
+//
+
+import UIKit
+
+class RestaurantCell: UITableViewCell{
+    static let identifier = "RestaurantCellIdentifier"
+        
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?){
+        super.init(style: .subtitle, reuseIdentifier: reuseIdentifier)
+        setup()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private let restaurantLogo: UIImageView = {
+        let logoImageView = UIImageView()
+        logoImageView.image = #imageLiteral(resourceName: "restaurant-logo")
+        logoImageView.translatesAutoresizingMaskIntoConstraints = false
+        logoImageView.contentMode = .scaleAspectFill
+        logoImageView.layer.cornerRadius = 23
```

Comment:
Talvez pegar esse valor do radius baseado no frame do elemento garanta que em diferentes tamanhos e resoluções esse componente nunca ficará distorcido.
```suggestion
        logoImageView.layer.cornerRadius = logoImageView.frame.width / 2
```

Reply:
Nesse caso eu tenho um problema, o frame tá com o valor zerado, imagino que o valor dos constraints de height e 
e width não estão chegando ainda pela ordem do fluxo. Qual a solução pra essa situação?

Reply:
Podemos definir nosso elemento como uma `lazy var`, dessa forma ela só será construída no momento em que é utilizada (chamada), nesse momento já teremos nosso frame com os valores corretos.
Uma outra forma também é fazer a atribuição do corner radius no método `extraSetupConfiguration` do protocolo do ViewCode.

Reply:
Não consegui fazer funcionar com o lazy var mas foi pelo setupExtraConfiguration()
essa seria a melhor forma?

```suggestion
        func setupExtraConfiguration() {
         restaurantLogo.layer.cornerRadius = self.frame.height / 2
    }
```

Reply:
Acredito que seja a necessidade se setar a flag `maskToBounds` da layer para true na construção do elemento.
```swift
logoImageView.layer.masksToBounds = true
```

https://developer.apple.com/documentation/quartzcore/calayer/1410896-maskstobounds
_"When the value of this property is true, Core Animation creates an implicit clipping mask that matches the bounds of the layer and includes any corner radius effects. If a value for the mask property is also specified, the two masks are multiplied to get the final mask value.
The default value of this property is false."_

Reply:
Pronto, funcionou certinho, coloquei também o masksToBounds. Obrigado